### ISSUE TITLE
[floats] missing sign of minus zero in listings view, truncated precision in decompiler view

### ISSUE THREAD

**Issue by intrigus-lgtm** [NationalSecurityAgency/ghidra#6677]
**Describe the bug**
1. Ghidra does not properly show the sign of minus zero (0.0f) (it's 0x8000_0000):
Listing view (incorrect):
![grafik](https://github.com/NationalSecurityAgency/ghidra/assets/60750685/17d0764a-a0ed-4fac-bfc0-7292952ae3d6)
Decompiler view (correct):
![grafik](https://github.com/NationalSecurityAgency/ghidra/assets/60750685/46c350d0-6bb7-44b0-af67-b688755eb029)

3. Ghidra truncates floats incorrectly in the decompiler view only:
Listing view (correct):
![grafik](https://github.com/NationalSecurityAgency/ghidra/assets/60750685/d5fcc6cc-670e-45ce-8322-39052370b5d1)
Decompiler view (incorrect):
![grafik](https://github.com/NationalSecurityAgency/ghidra/assets/60750685/4cf05cb2-285d-41e2-b543-2a28d087074c)

**To Reproduce**
1. Compile this code:
```c

float minus_zero;
float plus_zero;
float positive_infinity;
float negative_infinity;
float precision_problem;

int main(int argc, char **argv)
{

    minus_zero = -0.0f;
    plus_zero = 0.0f;
    positive_infinity = 1 / 0.0f;
    negative_infinity = -1 / 0.0f;
    precision_problem = 2.4912346E12f;
    return 0;
}
```
2. `gcc float.c -o float` or similar
4. Import the binary into ghidra.
5. Give the float type to the constants.
6. Observe that minus zero is missing its minus sign in the listings view (only) and that the precision of `2.4912346E12f` is wrong in the decompiler view (only).

**Expected behavior**
1. Minus sign is not missing in the listings view
2. `2.4912346E12f` is shown correctly in the decompiler view.

**Attachments**
[float.tar.gz](https://github.com/user-attachments/files/16016168/float.tar.gz)

**Environment (please complete the following information):**
Ghidra Version: 11.1.1
Ghidra Release: PUBLIC
Ghidra Build Date: 2024-Jun-14 1025 EDT
Ghidra Revision: febbeb447af1f059d583b11d7cefc8758b99f887
Ghidra Development Mode: false
OS Name: Linux
OS Arch: amd64
OS Version: 6.1.0-21-amd64
OS Pretty Name: Debian GNU/Linux 12 (bookworm)
Java Vendor: Eclipse Adoptium
Java Version: 22.0.1

**Additional context**
Originally found in the context of Google CTF's ieee challenge.

**Comment by ghidra1** [comment]
Fix is in the works for 11.1.2

**Comment by ghidra1** [comment]
@intrigus-lgtm Could you please create a separate issue for Decompiler formatting concern.  This ticket will be closed when the assembly formatting of -0.0 is merged.  Thx

**Comment by intrigus-lgtm** [comment]
@ghidra1 [done](https://github.com/NationalSecurityAgency/ghidra/issues/6708), thanks for fixing :tada:
